contract_api/registry.py

```diff
@@ -0,0 +1,372 @@
+from pymysql import MySQLError
+from contract_api.filter import Filter
+
+
+class Registry:
+    def __init__(self, obj_repo):
+        self.repo = obj_repo
+        self.obj_utils = Utils()
+
+    def _get_all_service(self):
+        try:
+            all_orgs_srvcs_raw = self.repo.execute("SELECT O.org_id, O.owner_address, S.service_id  FROM service S, "
+                                                   "organization O WHERE S.org_id = O.org_id AND S.is_curated = 1")
+            all_orgs_srvcs = {}
+            for rec in all_orgs_srvcs_raw:
+                if rec['org_id'] not in all_orgs_srvcs.keys():
+                    all_orgs_srvcs[rec['org_id']] = {'service_id': [],
+                                                     'owner_address': rec['owner_address']}
+                all_orgs_srvcs[rec['org_id']]['service_id'].append(
+                    rec['service_id'])
+            return all_orgs_srvcs
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _get_all_members(self):
+        try:
+            all_orgs_members_raw = self.repo.execute(
+                "SELECT org_id, member FROM members M")
+            all_orgs_members = defaultdict(list)
+            for rec in all_orgs_members_raw:
+                all_orgs_members[rec['org_id']].append(rec['member'])
+            return all_orgs_members
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def get_all_org(self):
+        """ Method to get list and high level details of all organization."""
+        try:
+            all_orgs_srvcs = self._get_all_service()
+            all_orgs_members = self._get_all_members()
+            all_orgs_data = []
+            for rec in all_orgs_srvcs:
+                data = {"org_id": rec,
+                        "owner_address": all_orgs_srvcs[rec]['owner_address'],
+                        "service_id": all_orgs_srvcs[rec]['service_id'],
+                        "members": all_orgs_members.get(rec, [])}
+                all_orgs_data.append(data)
+            return all_orgs_data
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _prepare_subquery(self, s, q, fm):
+        try:
+            sub_qry = ""
+            if s == "all":
+                for rec in fm:
+                    sub_qry += fm[rec] + " LIKE '" + str(q) + "%' OR "
+                sub_qry = sub_qry[:-3] if sub_qry.endswith("OR ") else sub_qry
+            else:
+                sub_qry += fm[s] + " LIKE '" + str(q) + "%' "
+            return sub_qry.replace("org_id", "M.org_id")
+        except Exception as err:
+            raise err
+
+    def _get_total_count(self, sub_qry, filter_query, values):
+        try:
+            if filter_query != "":
+                filter_query = " AND " + filter_query
+            search_count_query = "SELECT count(*) as search_count FROM service A, (SELECT DISTINCT M.org_id, M.service_id FROM " \
```

Comment:
Why are we joining with Service on the count query?

Reply:
To get the count of curated service only. we maintain is_curated field in service table.